csrc/python_frontend/fusion_record.h

```diff
@@ -932,6 +933,87 @@ struct CastOpRecord : RecordFunctor {
   PrimDataType dtype_;
 };
 
+struct CatOpRecord : RecordFunctor {
+  CatOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      int64_t dim)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.cat",
+            RecordType::CatOp),
+        dim_(dim) {}
+  virtual ~CatOpRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new CatOpRecord(*this);
+  }
+
+  virtual size_t hash() const final {
+    auto result = RecordFunctor::hash();
+    return result | (static_cast<size_t>(dim_) & 0xffff);
+  }
+
+  virtual bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (auto child_ptr = dynamic_cast<const CatOpRecord*>(&other)) {
+      result = RecordFunctor::operator==(other) && dim_ == child_ptr->dim_;
+    }
+    return result;
+  }
+
+  void operator()(FusionState& fd) final {
+    std::vector<TensorView*> input_tvs;
+    input_tvs.reserve(args_.size());
+    for (auto& a : args_) {
+      input_tvs.push_back(
+          fd.getFusionState(a.index)->template as<TensorView>());
+    }
+    auto output = cat(input_tvs, dim_);
+    fd.setFusionState(outputs_.at(0).index, output);
+  }
+
+  void print(std::ostream& os, bool close_function = true) const final {
```

Comment:
ah. so this is because RecordFunctor can't handle vector of states?

Reply:
Yeah exactly. The default print() assumes all `inputs` were passed directly to the op, but unless we switch the behavior of `cat` to take in `*args` instead of a single list of tensors, that doesn't work.